### ISSUE TITLE
Template IPYNB – force scikit learn UndefinedMetricWarning as error

### ISSUE THREAD

**Issue by mattjw** [mattjw/scraps#1]
Warning when an evaluation method cannot be evaluated due to missing class in y_train or y_test. Silent or semi-silent metric warnings unwanted. These should force error.

Edit this file: https://github.com/mattjw/scraps/blob/master/IPython/workbench_general.ipynb

Add this:

```
import warnings
#from sklearn.exceptions import UndefinedMetricWarning
from sklearn.metrics.base import UndefinedMetricWarning  # old API
warnings.filterwarnings("error", category=UndefinedMetricWarning)
```

**Comment by mattjw** [comment]
Additional note: sklearn's behaviour is to warn and set these metrics to zero. It is very incorrect to interpret these as natural zeros.

**Comment by mattjw** [comment]
Add statsmodels:

```
import statsmodels
from statsmodels.graphics.tsaplots import plot_acf
from statsmodels.tsa.arima_model import ARIMA
from statsmodels.tsa.seasonal import seasonal_decompose
from statsmodels.tsa.stattools import adfuller
import statsmodels.api as sm
```

More warnings:

```
import warnings
#from sklearn.exceptions import UndefinedMetricWarning

from sklearn.metrics.base import UndefinedMetricWarning  # old API
#warnings.filterwarnings("error", category=UndefinedMetricWarning)
warnings.filterwarnings("always", category=UndefinedMetricWarning)

from statsmodels import ConvergenceWarning
warnings.filterwarnings("error", category=ConvergenceWarning)
```

**Comment by mattjw** [comment]
Improved plot styling. Academic paper style, rather than ggplot nonsense.

```
plt.style.use('seaborn-white')  # set up white presets
plt.style.use('seaborn-deep')   # update with nicer colors
mpl.rcParams.update({
    # spines (lines), but does not control ticks
    # see: ax.xaxis.set_ticks_position('bottom')
    'axes.spines.top':      True,
    'axes.spines.left':     True,
    'axes.spines.bottom':   True,
    'axes.spines.right':    True,
    # x
    'xtick.major.size':     4.5,
    'xtick.minor.size':     2.5,
    'xtick.minor.visible':  True, 
    'xtick.major.width':    1.0,
    'xtick.minor.width':    0.5,
    # y
    'ytick.major.size':     4.0,
    'ytick.minor.size':     2.0,
    'ytick.minor.visible':  True, 
    'ytick.major.width':    1.0,
    'ytick.minor.width':    0.5,
})
```

**Comment by mattjw** [comment]
```
mpl.rc('savefig', dpi=500, format='pdf', bbox='tight')
```
